sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,223 @@
+	rpName: string;
+}
+type DeepPartialConfigKeys = 'rp' | 'user' | 'authenticatorSelection';
+
+type DeepPartial<T> = T extends object
+	? {
+			[P in keyof T]?: DeepPartial<T[P]>;
+		}
+	: T;
+
+type BrowserPasswordProviderOptions = Pick<
+	DeepPartial<PublicKeyCredentialCreationOptions>,
+	DeepPartialConfigKeys
+> &
+	Omit<
+		Partial<PublicKeyCredentialCreationOptions>,
+		DeepPartialConfigKeys | 'pubKeyCredParams' | 'challenge'
+	>;
+
+export interface PasskeyProvider {
+	create(): Promise<RegistrationCredential>;
+	get(challenge: Uint8Array): Promise<AuthenticationCredential>;
+}
+
+// Default browser implementation
+export class BrowserPasskeyProvider implements PasskeyProvider {
+	#name: string;
+	#options: BrowserPasswordProviderOptions;
+
+	constructor(name: string, options: BrowserPasswordProviderOptions) {
+		this.#name = name;
+		this.#options = options;
+	}
+
+	async create(): Promise<RegistrationCredential> {
+		return (await navigator.credentials.create({
+			publicKey: {
+				timeout: 60000,
+				...this.#options,
+				rp: {
+					name: this.#name,
+					...this.#options.rp,
+				},
+				user: {
+					name: this.#name,
+					displayName: this.#name,
+					...this.#options.user,
+					id: randomBytes(10),
+				},
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
+				pubKeyCredParams: [{ alg: -7, type: 'public-key' }],
+				authenticatorSelection: {
+					...this.#options.authenticatorSelection,
+					authenticatorAttachment: 'cross-platform',
+					residentKey: 'required',
+					requireResidentKey: true,
+					userVerification: 'required',
+				},
+			},
+		})) as RegistrationCredential;
+	}
+
+	async get(challenge: Uint8Array): Promise<AuthenticationCredential> {
+		return (await navigator.credentials.get({
+			publicKey: {
+				challenge,
+				userVerification: 'required',
+				timeout: this.#options.timeout,
+			},
+		})) as AuthenticationCredential;
+	}
+}
+
+/**
+ * @experimental
+ * A passkey signer used for signing transactions. This is a client side implementation for [SIP-9](https://github.com/sui-foundation/sips/blob/main/sips/sip-9.md).
+ */
+export class PasskeyKeypair extends Signer {
+	private publicKey: Uint8Array;
+	private provider: PasskeyProvider;
+
+	/**
+	 * Get the key scheme of passkey,
+	 */
+	getKeyScheme(): SignatureScheme {
+		return 'Passkey';
+	}
+
+	/**
+	 * Creates an instance of Passkey signer. It's expected to call the static `getPasskeyInstance` method to create an instance.
+	 * For example:
+	 * ```
+	 * const signer = await PasskeyKeypair.getPasskeyInstance();
+	 * ```
+	 */
+	constructor(publicKey: Uint8Array, provider: PasskeyProvider) {
+		super();
+		this.publicKey = publicKey;
+		this.provider = provider;
+	}
+
+	/**
+	 * Creates an instance of Passkey signer invoking the passkey from navigator.
+	 */
+	static async getPasskeyInstance(
+		provider: PasskeyProvider = new BrowserPasskeyProvider('Sui Wallet', {
```

Comment:
I think @Jordan-Mysten was suggesting to remove this default here, I don't have a strong preference one way or the other

Reply:
actually, with the `name` parameter required, I think removing the default definitely makes sense